core/events_bubble_open.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Events fired as a result of bubble open.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+'use strict';
+
+goog.provide('Blockly.Events.BubbleOpen');
+
+goog.require('Blockly.Events');
+goog.require('Blockly.Events.Ui');
+goog.require('Blockly.registry');
+goog.require('Blockly.utils.object');
+
+/**
+ * Class for a bubble open event.
+ * @param {Blockly.BlockSvg} opt_block The associated block. Undefined for a
+ *    blank event.
+ * @param {boolean=} opt_isOpen Whether the bubble is opening (false if closing).
+ * @param {string=} opt_element The type of bubble. One of 'mutator', 'comment'
+ *    or 'warning'
+ * @extends {Blockly.Events.Ui}
+ * @constructor
+ */
+Blockly.Events.BubbleOpen = function(opt_block, opt_isOpen, opt_element) {
+  var workspaceId = opt_block ? opt_block.workspace.id : undefined;
+  Blockly.Events.BubbleOpen.superClass_.constructor.call(this, workspaceId);
+  this.blockId = opt_block ? opt_block.id : null;
+
+  /**
+   * Whether the bubble is opening (false if closing).
+   * @type {boolean|undefined}
+   */
+  this.isOpen = opt_isOpen;
+
+  /**
+   * The type of bubble. One of 'mutator', 'comment', or 'warning'.
+   * @type {string|undefined}
+   */
+  this.element = opt_element;
+};
+Blockly.utils.object.inherits(Blockly.Events.BubbleOpen, Blockly.Events.Ui);
+
+/**
+ * Type of this event.
+ * @type {string}
+ */
+Blockly.Events.BubbleOpen.prototype.type = Blockly.Events.BUBBLE_OPEN;
```

Comment:
Why are we inheriting from UI events, if we're overriding the type. 

If someone wants to filter for all events that are not UI events, then they won't be able to just do this no more: 
```
if (event.type !== 'ui') {
 ...
}
```

Reply:
UI events have a new property `IS_UI_EVENT`, in order to filter UI events, one can:
```
if (event.IS_UI_EVENT) {
  ...
}
```

Reply:
Ah got it.